mcs/resource_mcs_bs_volume.go

```diff
@@ -0,0 +1,325 @@
+package mcs
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	bsVolumeCreateTimeout = 30 * time.Minute
+	bsVolumeDelay         = 10 * time.Second
+	bsVolumeMinTimeout    = 3 * time.Second
+)
+
+var (
+	bsVolumeStatusBuild    = "creating"
+	bsVolumeStatusActive   = "available"
+	bsVolumeStatusShutdown = "deleting"
+	bsVolumeStatusDeleted  = "deleted"
+)
+
+func resourceBlockStorageVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBlockStorageVolumeCreate,
+		Read:   resourceBlockStorageVolumeRead,
+		Update: resourceBlockStorageVolumeUpdate,
+		Delete: resourceBlockStorageVolumeDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(bsVolumeCreateTimeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+
+			"size": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"volume_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+
+			"metadata": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: false,
+			},
+
+			"snapshot_id": {
```

Comment:
Is it really required to keep next three fields? After the volume had been built the source may be deleted so a source ref became a broken item.

Reply:
We need to keep all fields in terraform state, though we will not use there three after volume has been created

Reply:
ok